core/templates/pages/exploration-editor-page/editor-tab/services/responses.service.ts

```diff
@@ -212,19 +210,11 @@ export class ResponsesService {
   // The 'data' arg is a list of interaction handlers for the
   // currently-active state.
   init(data: Interaction): void {
-    this.answerGroupsCacheService.reset();
```

Comment:
Why are we no longer using the cache service?

Reply:
I recently found a bug in the codebase that I believe has been deployed for a while now. The bug occurs in ResponsesService.ts, specifically with AnswerGroupsCacheService. 

The bug is reproduced as follows. It works on the currently deployed version and on develop (see video [here](https://youtu.be/UYcjHS2d1Cg)):
1. Create an ItemSelectionInput (any interaction works, just using ItemSelectionInput as an example)
2. Create an AnswerGroup.
3. Refresh the page.
4. Delete the interaction, and create an ItemSelectionInput again.

The observed behavior will be that the answer group you created before has returned. This results in an old answer group being saved, even though it may not match the new choices. It turns out that refreshing the page causes ResponsesService to initialize itself, and the cached answer groups are saved in the AnswerGroupsCacheService. If you were not to refresh the page, this issue would not occur as the AnswerGroupsCacheService would always store an empty array.

AnswerGroupsCacheService seems to either (1) always cache an empty array (defeating the purpose of a cache) or (2) always cache an potentially invalid/previous answer group, which made me decide to remove it.

Reply:
oh! Nice catch @shavavo and thanks a lot for fixing this.